inject-tck-runner/src/test/java/org/jboss/weld/atinject/tck/Producers.java

```diff
@@ -63,8 +65,10 @@ public Seat produceAtDriversSeat(@New DriversSeat driversSeat) {
      */
     @Produces
     @Typed(DriversSeat.class)
-    public DriversSeat produceDriversSeat(@New DriversSeat driversSeat) {
-        return driversSeat;
+    public DriversSeat produceDriversSeat(BeanManager manager) {
+        // TODO this used to use @New to create the instance, do this in a cleaner way
```

Comment:
I am not even sure how this worked with `@New` in the first place since this was basically a circular dependency.

While the `Unmanaged` approach here would probably work, it is meant just as a hack to get a compile-able code.
It needs to be rewritten to something reasonable - maybe creating the instance via BM and injecting into it or an extension.
Same scenario applies on L#100 of this class.

I'd be curious to hear how this worked and what  you suggest instead @mkouba :)